### DISCUSSION TITLE
Who uses CacULE?

### DISCUSSION THREAD

**Discussion by hamadmarri** [hamadmarri/cacule-cpu-scheduler#30] [General]
In this topic I will post some links I found where CacULE is used. Also, you too could introduce yourself and tell us about your experience with Cachy/CacULE.

The reasons why I am collecting the sources where CacULE is used are:

1- To make a reference for people who just discovered CacULE, so to give the scheduler more credits when they see it has been used by different custom kernels or patches.

2- The vise-versa of point one, there are many good projects where people don't know yet. In case if someone cared about CacULE and reading this topic, it is also worthy to show them others projects and patch sets  where CacULE is used.

3- Skip point three since it has low interactivity score.

4- For my career. Some job recruiters like to see if my project is really in use.

Thank you

**Comment by hamadmarri** [comment]
XanMod (https://xanmod.org)

The first hand supported Cachy/CacULE since the beginning. By support I mean coding, testing, and adopting.

https://xanmod.org
https://www.reddit.com/r/XanMod/

Many thanks to Alexandre Frade.

**Comment by hamadmarri** [comment]
GentooX
>I found it rather tedious to setup a clean Gentoo install and LTO all the packages, not only gcc has to be rebuilt to support graphite, but then your entire stage3 install needs to be recompiled, not to mention already installed software. GentooX aims to provide pre-compiled and LTOed packages from the get go with easy installation and convenient LiveCD. Since GentooX mandates AVX support, this allows further optimizations to all packages. 

https://gentoox.cryptohash.nl/
https://github.com/fatalhalt/gentoox/commit/fb44bd22fcf1e8de5865360ff61e82853cb75e1b

**Comment by hamadmarri** [comment]
http://gpo.zugaina.org/sys-kernel/xanmod-cacule-sources

**Comment by hamadmarri** [comment]
In AUR
https://aur.archlinux.org/packages/linux-xanmod-cacule/
https://aur.archlinux.org/packages/linux-xanmod-cacule-uksm/
https://aur.archlinux.org/packages/linux-manjaro-xanmod-cacule-uksm/
https://aur.archlinux.org/packages/linux-xanmod-cacule-uksm-cjktty/

**Comment by hamadmarri** [comment]
Fedora Copr
https://copr.fedorainfracloud.org/coprs/rmnscnce/kernel-xanmod/

**Comment by hamadmarri** [comment]
Opensuse OBS
https://build.opensuse.org/package/show/home:nhubaotruong/linux-xanmod-cacule-uksm-cjktty
https://build.opensuse.org/package/show/home:Shinya/Xanmod-cachy

**Comment by ptr1337** [comment]
I think one of the first who really found and used your scheduler was  @kevall474 (https://github.com/kevall474/Linux) 
I've found his first kernel builder like 4-5 months ago, which where on another account, i need to do some research to find that. 
At this time it were called "Cachy", after my first kernel build i was fascinated how smooth and how less input lag i got in my favorite games, since then i only use the scheduler. I have tested some other from time to time, because i thought its maybe placebo, but i switched instantly back. 

After xanmod released his first kernel, there are more and more packages in the repository's, which is great. 
Just wanted to mention him, since he is doing great work. 

And a really big thanks to you, you have improved my linux experience much! Great work keep on! 
Something like that scheduler should replace the standard cfs scheduler. 

Regards.

**Reply by hamadmarri** [reply]
Yes, I was to add @kevall474 repo :+1: 

Thank you so much @ptr1337

**Comment by hamadmarri** [comment]
Some discussions and feedback:

by FPScholten
>Just reporting that I switched to the cacule version of xanmod and so far it runs flawless.
It does not have the issues with CPU intensive tasks that cachy had. Everything runs smooth. Surprisingly HEVC video encoding was (marginally) faster with cacule than with cfs while the system remained responsive! 
https://forum.xanmod.org/thread-2-post-7392.html#pid7392

Some analysis by firelzrd
https://forum.xanmod.org/thread-2-post-7420.html#pid7420

Cacule rules! by dyna
https://forum.xanmod.org/thread-4076.html?highlight=cacule

https://twitter.com/oddfox/status/1355647758040875008

**Comment by hamadmarri** [comment]
Github repos

https://github.com/HougeLangley/Xanmod-CaCule-UKSM
https://github.com/HougeLangley/customkernel

https://github.com/kevall474/Linux

https://github.com/pcmid/linux-xanmod-cacule-uksm

https://github.com/baka-world/archlinux-xanmod-cacule

https://github.com/zhullyb/linux-xanmod-cacule-uksm-cjktty

https://github.com/Jebaitedneko/linux-manjaro-xanmod

https://github.com/Frogging-Family/community-patches

https://github.com/SveSop/kernel_cybmod

https://github.com/sandikata/ROKO__

https://github.com/nullxception/linux-chaldeans

https://github.com/phush0/patches510

https://github.com/redrootmin/bzu-gmb

EDIT:
Gitlab repos
https://gitlab.com/raykzhao/raykzhao-overlay/-/tree/master/sys-kernel/ray-cacule-sources

Sorry if I forgot any.

**Reply by raykzhao** [reply]
Hi @hamadmarri 

My custom kernel is at my personal overlay for Funtoo. The CacULE patch is under the `files` folder.  https://gitlab.com/raykzhao/raykzhao-overlay/-/tree/master/sys-kernel/ray-cacule-sources

**Reply by hamadmarri** [reply]
Thank you @raykzhao :+1: :+1:

**Comment by hamadmarri** [comment]
draugeros
They use XanMod by default. They also included xanmod-cacule in their repo

https://draugeros.org/go/
https://apt.draugeros.org/pool/main/l/

**Comment by dim-geo** [comment]
Hello, first of all congrats for your work.
I have used it with xanmod for a week so far, under gentoo with no issues at all, even under cpu/io load.

TBH, I use the xanmod kernel config which differs significantly with your suggested config. Can you please check with xanmod why is that?

**Reply by hamadmarri** [reply]
Hi @dim-geo 

Most of my suggested configs are to soften the security to gain slightly more performance. XanMod is general purpose kernel and many non-tech people use it, therefore, I would not recommend such configs in xanmod.

Thank you

**Comment by AstroProfundis** [comment]
I've been using it in my custom kernel (on ArchLinux) for few months. The repo is not published to AUR as I randomly change things and don't want to break others, but there are indeed some of my friends also using this kernel for daily usages.

I'm using the CacULE patch on top of the zen-kernel tree, so some minor adjustment for Kconfig was made (the `!SCHED_MUQSS` part has conflict with `!SCHED_CACULE`, pure text issues). The kernel is running on my laptop and my home lab building server, and so far it was working very well, I didn't noticed any problem that may be caused by the scheduler.

**Reply by mabod** [reply]
linux-zen is my default kernel on Arch. Can you share what you did to add the cacule scheduler to it, please?

**Comment by hamadmarri** [comment]
aarch64/armv7:
https://aur.archlinux.org/packages/linux-raspberrypi4-cacule-stable/
https://aur.archlinux.org/packages/linux-raspberrypi4-cacule-lts/

x86_64/amd64:
https://aur.archlinux.org/packages/linux-cacule/
https://aur.archlinux.org/packages/linux-cacule-rdb/

linux-raspberrypi4-cacule-stable 5.11.11-2
https://aur.archlinux.org/packages/linux-raspberrypi4-cacule-stable/

Thanks to @ptr1337

**Comment by ptr1337** [comment]
Ive created now for archlinux x86_64 and armv7/aarch64 one repo, so that everything is on one place.
Also created a readme for generel informations to improve the scheduler. Will be worked out these days more! 
x86_64:
linux-cacule (with several other patches) Kernel 5.12 and 5.11
linux-cacule-rdb (still experimental!! with several other patches) Kernel 5.12 and 5.11
linux-hardened-cacule (Hardened Kernel with cacule as scheduler) Kernel 5.11

aarch64/armv7h:
linux-raspberrypi4-cacule (RDB can be enabled in the config, build is working for both architectures) 5.10.32
linux-raspberrypi4-cacule-stable (RDB can be enabled in the config, build is working for both architectures) 5.11.15

I also upload builded packages to my fileserver, so you dont need to compile them if you want to test:
https://ptr1337.dev

GitHub:
https://github.com/ptr1337/linux-cacule-aur

Regards!
